main/Main.hs

```diff
@@ -229,20 +221,20 @@ mergeToolJob opts@Nixfmt{files = [base, local, remote, merged]} = runExceptT $ d
               <$> formatTarget formatter (fileTarget path)
         )
 
-  (_, Just out, _, process) <- do
-    lift $
-      createProcess
-        (proc "git" ["merge-file", "--stdout", local, base, remote])
-          { std_out = CreatePipe
-          }
+  exitCode <- lift $ withFile merged WriteMode $ \out -> do
```

Comment:
Feel free to ignore this comment:

My haskell is too weak to really grok this code. Is there any issue if we crash partway through invoking `git merge-file`? Could we end up with a cut off file?

I was expecting to see something like "write to a temp file and then atomically move". But I guess we didn't have that in the old code, or at least I don't see anything like that.

Reply:
Good point! I now added a commit that both simplifies the code and takes care of this. It works by not using the `--stdout` flag, causing `git merge-file` to write it to the `local` file directly (which is a temporary file created by `git mergetool`), and then move that file to `merged` after formatting :D